src/System.Management.Automation/engine/Attributes.cs

```diff
@@ -1284,18 +1284,25 @@ protected override void Validate(object arguments, EngineIntrinsics engineIntrin
                     null, Metadata.ValidateCountNotInArray);
             }
 
+            if (MinLength == MaxLength && len != MaxLength)
```

Comment:
@lzybkr 
I think the fundamental question is: Is it a _design goal_ to provide prescriptive guidance? If it is (I think it should be), then let's _fix_ and improve the currently _broken_ / incomplete guidance - and that would mean fixing the 2 broken messages and adding one for the exact-count case.

There are limits to how helpful a _generic_ validation error message can be (see #3630), but here's an opportunity to be as helpful as possible.

Personally, if I specified an incorrect number of values to a parameter that accepts an exact number and I got the following message, I'd feel like I'm being presented with a brain-teaser rather than a helpful error message:
`The parameter requires at least 2 value(s) and no more than 2 value(s).`